functions/APIs/queries.js

```diff
@@ -0,0 +1,95 @@
+const { db } = require('../util/admin');
+const axios = require('axios');
+require('dotenv').config();
+
+/*
+find food banks that are a specific distance from a farm
+GET /queryFoodBanksByDistance
+URL params: locationId, distance
+success response: array of objects
+*/
+exports.queryFoodBanksByDistance = (request, response) => {
+    db.collection('foodbanks').get()
+    .then((data) => {
+        let foodbanks = [];
+        data.forEach((doc) => {
+            foodbanks.push(doc.data());
+        });
+
+        function calculateDistances(data) {
+            return new Promise(function(resolve, reject) {
+                axios.get(`https://maps.googleapis.com/maps/api/distancematrix/json?units=imperial&origins=place_id:${request.query.locationId}&destinations=place_id:${data.locationId}&key=${process.env.API_KEY}`)
```

Comment:
could be helpful to save this string in a few variables then concatenate it in the axios get request so it is easier to read

Reply:
Good point, Andrew.  You can also build URLs in a more object-oriented way in JavaScript.  Check this out:

```
const requestUrl = new URL("https://maps.googleapis.com/maps/api/distancematrix/json");
requestUrl.searchParams.set('units', 'imperial');
requestUrl.searchParams.set('origins', `place_id:${request.query.locationId}`);
requestUrl.searchParams.set('destinations', `place_id:${data.locationId}`);
requestUrl.searchParams.set('key', process.env.API_KEY);
axios.get(requestUrl.href);
```

In fact, since you use this same URL in multiple places, why not factor out a helper function that does this and returns the result?  That way you won't have to repeat the parts that are constant.